Formula/libngspice.rb

```diff
@@ -0,0 +1,49 @@
+class Libngspice < Formula
+  desc "Spice circuit simulator as shared library"
+  homepage "http://ngspice.sourceforge.net"
+  url "https://downloads.sourceforge.net/project/ngspice/ng-spice-rework/26/ngspice-26.tar.gz"
+  sha256 "51e230c8b720802d93747bc580c0a29d1fb530f3dd06f213b6a700ca9a4d0108"
+  head "git://ngspice.git.sourceforge.net/gitroot/ngspice/ngspice"
+
+  if build.head?
+    depends_on "autoconf" => :build
+    depends_on "automake" => :build
+    depends_on "bison" => :build
+    depends_on "libtool" => :build
+  end
+
+  option "without-xspice", "Build without x-spice extensions"
+
+  def install
+    if build.head?
+      system "./autogen.sh"
+    end
+
+    args = %W[
+      --disable-debug
+      --disable-dependency-tracking
+      --prefix=#{prefix}
+      --with-editline=yes
+      --with-ngshared
+    ]
+    args << "--enable-xspice" if build.with? "xspice"
+
+    system "./configure", *args
+    system "make", "install"
+  end
+
+  test do
+    (testpath/"test.cpp").write <<-EOS.undent
+      #include <cstdlib>
+      #include <sharedspice.h>
+      int ng_exit(int status, bool immediate, bool quitexit, int ident, void *userdata) {
+        return status;
+      }
+      int main() {
+        return ngSpice_Init(NULL, NULL, ng_exit, NULL, NULL, NULL, NULL);
+      }
+    EOS
+    system ENV.cc, "test.cpp", "-I#{share}/ngspice/include/ngspice", "-L#{lib}", "-lngspice", "-o", "test"
```

Comment:
After researching this issue a bit, I see several options to resolve it:

1. Symlink the folder `share/ngspice/include/ngspice` to `include/ngspice` by `include.install_symlink Dir[share/"ngspice/include/*"`
2. Use a patch file from the [upstream ticket] that modifies the makefiles to install the headers into the correct location.

What is the preferred way to do?

[upstream ticket]:https://sourceforge.net/p/ngspice/bugs/327/